Provider/Pool.php

```diff
@@ -213,23 +233,33 @@ public function getProviderList()
     }
 
     /**
-     * @param \Sonata\MediaBundle\Model\MediaInterface $media
+     * @deprecated since version 2.4, to be removed in 3.0.
+     */
+    public function getDownloadSecurity(MediaInterface $media)
+    {
+        @trigger_error('The '.__METHOD__.' method is deprecated since version 2.4 and will be removed in 3.0.', E_USER_DEPRECATED);
+        
+        return array_merge($this->getDownloadSecurity($media), $this->getDownloadStrategy($media));
```

Comment:
IMO Calling ``array_merge`` is more BC, if you add something directly to the ``$this->downloadSecurities`` without calling any method.